lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -263,6 +270,64 @@ deriveAddressPrivateKey (Passphrase pwd) accountKey idx@(Index addrIx) = RndKey
     , payloadPassphrase = payloadPassphrase accountKey
     }
 
+{-------------------------------------------------------------------------------
+                          Storing and retrieving keys
+-------------------------------------------------------------------------------}
+
+instance PersistKey RndKey where
+    serializeXPrv = serializeXPrvRnd
+    deserializeXPrv = deserializeXPrvRnd
+    deserializeXPub = deserializeXPubRnd
+    serializeXPub = serializeXPubRnd
+
+-- | Serialize the key with its payload encryption passphrase.
+serializeKey :: (ByteString, Passphrase "addr-derivation-payload") -> ByteString
+serializeKey (k, Passphrase p) =
+    convertToBase Base16 k <> ":" <> convertToBase Base16 p
+
+-- | Deserialize the key and its payload encryption passphrase.
+deserializeKey
+    :: (ByteString -> Either String key)
+    -> ByteString
+    -> Either String (key, Passphrase "addr-derivation-payload")
+deserializeKey f b = case map (convertFromBase Base16) (B8.split ':' b) of
+    [Right k, Right p] -> case f k of
+        Right k' -> Right (k', Passphrase (BA.convert p))
+        Left e -> Left e
+    _ -> Left "Key input must be two hex strings separated by :"
+
+serializeXPrvRnd
+    :: (RndKey depth XPrv, Hash "encryption")
```

Comment:
Wouldn't it be more correct / type-safe to have `RndKey 'RootK XPrv` here? Instead of leaving the depth parameter free. Since we are discarding the derivation path from the serialized value, it wouldn't make sense to allow serializing a `RndKey 'AddressK XPrv` via this method. Don't you think ?

Reply:
Yep, fixed.